compiler/rustc_trait_selection/src/solve/search_graph/mod.rs

```diff
@@ -158,8 +158,12 @@ impl<'tcx> SearchGraph<'tcx> {
             // Note that if any nested goal were to depend on something deeper on the stack,
             // this would have also updated the depth of the current goal.
             if depth == self.stack.next_index() {
-                for (i, entry) in cache.entries.drain_enumerated(provisional_entry_index.index()..)
+                for (entry, i) in cache
+                    .entries
+                    .drain(provisional_entry_index.index()..)
+                    .zip(provisional_entry_index.index()..)
```

Comment:
`i` was ranging from `0..` instead from `provisional_entry_index..`, because of the implementation of [`drain_enumerated`](https://doc.rust-lang.org/stable/nightly-rustc/src/rustc_index/vec.rs.html#206). Seems like a footgun in the `IndexVec` impl.

Reply:
I'll see if I can fix that instead